src/Ardalis.Specification/Builders/Builder_Flags.cs

```diff
@@ -0,0 +1,328 @@
+﻿namespace Ardalis.Specification;
+
+public static partial class SpecificationBuilderExtensions
+{
+    /// <summary>
+    /// Configures the specification to ignore query filters.
+    /// </summary>
+    /// <typeparam name="T">The type of the entity.</typeparam>
+    /// <typeparam name="TResult">The type of the result.</typeparam>
+    /// <param name="builder">The specification builder.</param>
+    /// <returns>The updated specification builder.</returns>
+    public static ISpecificationBuilder<T, TResult> IgnoreQueryFilters<T, TResult>(
+        this ISpecificationBuilder<T, TResult> builder) where T : class
+        => IgnoreQueryFilters(builder, true);
+
+    /// <summary>
+    /// Configures the specification to ignore query filters if the condition is true.
+    /// </summary>
+    /// <typeparam name="T">The type of the entity.</typeparam>
+    /// <typeparam name="TResult">The type of the result.</typeparam>
+    /// <param name="builder">The specification builder.</param>
+    /// <param name="condition">The condition to evaluate.</param>
+    /// <returns>The updated specification builder.</returns>
+    public static ISpecificationBuilder<T, TResult> IgnoreQueryFilters<T, TResult>(
+        this ISpecificationBuilder<T, TResult> builder,
+        bool condition) where T : class
+    {
+        if (condition)
+        {
+            builder.Specification.IgnoreQueryFilters = true;
+        }
+
+        return builder;
+    }
+
+    /// <summary>
+    /// Configures the specification to ignore query filters.
+    /// </summary>
+    /// <typeparam name="T">The type of the entity.</typeparam>
+    /// <param name="builder">The specification builder.</param>
+    /// <returns>The updated specification builder.</returns>
+    public static ISpecificationBuilder<T> IgnoreQueryFilters<T>(
+        this ISpecificationBuilder<T> builder) where T : class
+        => IgnoreQueryFilters(builder, true);
+
+    /// <summary>
+    /// Configures the specification to ignore query filters if the condition is true.
+    /// </summary>
+    /// <typeparam name="T">The type of the entity.</typeparam>
+    /// <param name="builder">The specification builder.</param>
+    /// <param name="condition">The condition to evaluate.</param>
+    /// <returns>The updated specification builder.</returns>
+    public static ISpecificationBuilder<T> IgnoreQueryFilters<T>(
+        this ISpecificationBuilder<T> builder,
+        bool condition) where T : class
+    {
+        if (condition)
```

Comment:
What's the purpose of passing in a condition flag?

Reply:
It's similar to WhereIf types of LINQ extensions. In our case, if the condition is false, the statement does nothing, meaning the expression/value is not added to the specification state. This allows you to have a more succinct way of expressing the query. Instead of having if clauses, you can pass that condition to the extension method itself. We have this for a long time.
The #420 contains more details